src/iterators/bfs.jl

```diff
@@ -0,0 +1,104 @@
+"""
+    BFSIterator
+
+`BFSIterator` is a subtype of [`VertexIterator`](@ref) to iterate through graph vertices using a breadth-first search. A source node(s) is optionally supplied as an `Int` or an array-like type that can be indexed if supplying multiple sources. If no source is provided, it defaults to the first vertex.
+    
+# Examples
+```julia-repl
+julia> g = smallgraph(:house)
+{5, 6} undirected simple Int64 graph
+
+julia> for node in BFSIterator(g,3)
+           display(node)
+       end
+3
+1
+4
+5
+2
+
+julia> for node in BFSIterator(g,[1,3])
+           display(node)
+       end
+1
+2
+3
+4
+5
+3
+1
+4
+5
+2
+```
+"""
+struct BFSIterator{S} <: VertexIterator
+    graph::AbstractGraph
```

Comment:
The type of the graph should be concrete, i.e., change to:
```suggestion
struct BFSIterator{S, T<:AbstractGraph{S}} <: VertexIterator
    graph::T
```